app/api/public-cloud/aws-roles/helpers.ts

```diff
@@ -40,6 +41,44 @@ const paginate = <T>(users: T[], options: PaginationOptions): T[] => {
   return users.slice(startIndex, endIndex);
 };
 
+const searchSubstringInArray = (searchTerm: string, users: Record<string, User>[]): Record<string, User>[] => {
+  const results = new Set<Record<string, User>>();
+
+  users.filter((userObj) =>
```

Comment:
Hmm, I'm curious about the data structure of 'users' since it involves nested arrays. In general, when the purpose is iterating over arrays and the 'filter' and 'some' functions are used to produce output, replacing them with 'forEach' can help avoid confusion.

Reply:
yes, forEach works perfectly, thanks